### PR TITLE
fix(home): replace real third-party brand defaults with RFC-2606 placeholders

### LINKED ISSUE #4340: 🔒 Home module — replace real-looking external brand defaults (RFC-2606)
The home module's social-proof defaults hardcode a **real-looking external brand + domain** (and a second real-looking domain) as shipped defaults of this **public** stack — a generic-OSS hygiene violation (defaults must not reference a real third party).

**Blast radius — 6 occurrence-pairs across 3 files (verified):**
- `src/config/index.js` (the **bundled runtime default** — the one that actually ships)
- `src/modules/home/config/home.development.config.js` (module config source)
- `src/modules/home/components/home.social.component.vue` (component default + comment example)

A component-only fix would close this while the bundled `src/config/index.js` keeps shipping the real domain — a half-fix. Fix **all** sites.

**Proposed:**
1. Replace every occurrence with an **RFC-2606 reserved** placeholder (`Example Co` / `https://example.com`) — never trade one real domain for another.
2. If `src/config/index.js` is generated from `src/config/defaults/*`, fix the **source** + regenerate (don't hand-edit the built file).
3. Add a **CI/`/verify` grep-guard** for hardcoded real brand/domain literals so it can't recur.
4. The same contamination class also appears in `lib/helpers/api.js:11` (comment) + an `api.unit.tests.js` fixture — fold into this checklist or file a sibling.

_Split from a docs-cleanliness audit (a sibling issue), 2026-06-16. Public-OSS hygiene — ship promptly._

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## What

The home module shipped two **real third-party brands** as social-proof defaults of this public stack — their names, their domains, and their actual logo image files. Defaults in a generic OSS stack must not reference a real third party.

- Names and links → RFC-2606 reserved placeholders (`Example Co` / `https://example.com`, `Example Corp` / `https://example.org`)
- `partner01.webp` / `partner02.webp` were the brands' **actual logo marks** (a "C" wordmark; a fox-head silhouette). Replaced with trivial neutral SVG placeholders; the `.webp` files are deleted
- Two test fixtures that used real famous domains as arbitrary external-origin values swapped for `example.com`
- New `scripts/check-brand-defaults.js`, wired into `npm run lint`, so this cannot recur silently

`src/config/index.js` is generated (and gitignored) from `src/config/defaults/*` plus the per-module `*.development.config.js`, so fixing the module config fixes the bundled default. Verified by regenerating and grepping.

## The guard, and what it actually covers

The guard went through three review rounds. Every hole below was found by **executing** the guard against a deliberately broken config, not by reading it — and each fix is proven red-before / green-after.

Closed over those rounds:

- `name` was never checked — only `link`. `name: 'Nike'` with a placeholder link passed clean.
- `src/config/defaults/*.config.js` was never scanned, yet it is the layer that **wins**: `generateConfig` merges it after the module config and `deepMerge` replaces arrays wholesale. A real brand injected there reached the served config while the guard reported clean.
- The doc-comment scan was anchored to the literal start of file. One leading newline (or a BOM) silently skipped it entirely. A missing comment or missing file also read as "clean".
- The local-link escape hatch was a `^[#/]` prefix test, so `//realbrand.com`, `/\realbrand.com` and `/<TAB>/realbrand.com` all passed while resolving off-site in a browser. Replaced with the real WHATWG URL parser against a sentinel origin, which closes the whole quirk family instead of enumerating forms. Unparseable input fails closed.
- `img` was never checked despite the docstring claiming logo coverage. A hotlinked third-party logo URL shipped clean.
- The error messages advised using `.edu`, which the regex rejects — following the guard's own advice produced a second failure.

**Honest limits, stated in the script itself:** it checks text. A real third-party logo committed as a *local* file under a neutral filename is invisible to it — which is exactly how this defect originally shipped, and why the image files were replaced by hand here. The `name` check is a prefix allowlist (`/^Example\b/`), so `Example Nike` would pass.

## Downstream consideration

After `/update-stack`, this guard runs in every consumer's `npm run lint`. Its file scan is deliberately scoped to the four env names this stack itself ships (`development`, `production`, `test`, `myproject`) so a consumer's own `{project}.config.js` overrides — where real partner data belongs — are never scanned. A consumer that instead put real partner data directly into the stack-identical `home.development.config.js` will see lint fail on their next sync; that file is meant to stay stack-identical under the ISO-merge rule.

## Verification

- `npm run lint` green (eslint + the new guard). Exit-code propagation confirmed through a raw shell, not the terminal wrapper.
- 2656 unit tests pass. Coverage 98.98 / 92.52 / 98.84 / 99.69 against thresholds 70/60/70/70 — unchanged, nothing lowered.
- `npm run build` exit 0; `dist/images/` contains the new SVGs and no leftover `.webp`.
- **Runtime visual check** in a real browser: both SVGs return **200**, `naturalWidth: 64` on all 8 rendered instances, **zero** requests for `partner*.webp` across 373 network requests, and no old brand string in the DOM, in any of 29 `href`s, or any of 18 `img` `src`s. No new console errors.
- E2E skipped — the Node API was not running locally. The change is a text/asset swap plus a build-time lint script; the visual check above covers the rendered surface.

## Review

Three independent cold-context review rounds pre-push (0 critical, 0 high at the final round) plus a security pass (no findings; the perimeter flag was `RegExp.prototype.exec`, not shell execution). The kimi-backed gate was unavailable, so the second model family for this PR is CodeRabbit post-PR.

Closes #4340

https://claude.ai/code/session_0185ELiCjZaBJx8PH4xoSsZb

## Summary by CodeRabbit

* **Documentation**
  * Updated social-proof configuration examples with generic company names, reserved example URLs, and SVG image references.

* **Chores**
  * Added automated validation to ensure shipped social-proof defaults and documented examples use safe placeholder values.
  * Integrated validation into the lint workflow.

* **Tests**
  * Updated URL fixtures to use reserved example domains without changing production behavior.
  * Expanded test discovery to include validation tests.

### CODE REVIEW

File: scripts/check-brand-defaults.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+const LOCAL_SENTINEL_HOST = new URL(LOCAL_SENTINEL).host;
+const isLocalLink = (link) => {
+  try {
+    return new URL(link, LOCAL_SENTINEL).host === LOCAL_SENTINEL_HOST;
+  } catch {
+    // Unparseable → fail closed: not local, so it still has to clear the
+    // RFC2606 check below to pass at all.
+    return false;
+  }
+};
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply scripts/check-brand-defaults.js:126]: Fixed in 16d8b1f5: added JSDoc headers to isLocalLink, isAllowedLink, isPlaceholderName, and main (one-line description + @param/@returns each).

File: scripts/check-brand-defaults.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  if (errors.length > 0) {
+    console.error('✗ Brand-default guard failed (Vue#4340):\n');
+    errors.forEach((e) => console.error(`  ${e}`));
+    console.error(`\n${LINK_ADVICE}\n${IMG_ADVICE}\n${NAME_ADVICE}`);
+    process.exit(1);
+  }
+  console.log('✓ Brand-default guard: home social-proof defaults are RFC-2606 clean.');
+};
+
+main();
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply scripts/check-brand-defaults.js]: Fixed in 16d8b1f5: added scripts/tests/check-brand-defaults.unit.tests.js (23 tests) covering valid/invalid config entries, the URL-parser local-link edge cases (protocol-relative //, backslash, embedded-tab quirks), the real shipped config/override files, and component doc-comment parsing (missing file, missing comment, brand literal, clean). Exported the guard's functions behind a main-module guard so importing the script for testing no longer runs main() as a side effect, and widened vitest's include glob to pick up scripts/**/tests/.

[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply scripts/check-brand-defaults.js]: Follow-up in b4f053dc: strengthened this per an explicit request to cover the specific defects each earlier manual review round found, not just the current-clean-state smoke tests from the previous commit. Now 33 tests, including: a real brand in `name` paired with an otherwise-valid link (must fail on its own); the `src/config/defaults` override layer that wins at merge (a wiring assertion on `CONFIG_SOURCES` + a fixture-backed mechanism test — fixture written inside the project tree since Vite's SSR module runner refuses file:// imports outside its root); `//host`, `/\host`, and embedded TAB/CR/LF URL-parser quirks; and doc-comment parsing across a leading newline, a leading BOM, a missing comment, and a missing file. Every one of these was verified to go red under a targeted mutation of the exact logic it covers (isPlaceholderName, isAllowedLink, isLocalLink, the CONFIG_SOURCES override-layer entry, and the doc-comment whitespace-tolerant regex), then restored to green — so none of it is decorative.
